### PR TITLE
Fix COM TypeDescriptor SetValue

### PR DESCRIPTION
**Pull request by JeremyKuhne**
The main purpose of the change was to start adding type descriptor tests for our COM descriptor (which is used by System.ComponentModel). Discovered that SetValue was broken, so the title is such.

This adds a starting set of tests, refactors IPicture/Disp helpers into a shared location, and references the outputs from AxHosts.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8246)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/PARAMTests.cs [hunk 1]
```diff
@@ -32,8 +32,11 @@ public unsafe void FromLowHighUnsigned_x32_Result()
         [ConditionalFact(typeof(ArchitectureDetection), nameof(ArchitectureDetection.Is64bit))]
         public unsafe void FromLowHighUnsigned_x64_Result()
         {
-            Assert.Equal((long)0x0000000003040102, (long)PARAM.FromLowHighUnsigned(0x0102, 0x0304));
-            Assert.Equal((long)0x00000000F3F4F1F2, (long)PARAM.FromLowHighUnsigned(0xF1F2, 0xF3F4));
+            if (Environment.Is64BitProcess)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/PARAMTests.cs:35]: Just silencing a failure in TestExplorer- it doesn't know to filter these conditions.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs [hunk 2]
```diff
@@ -942,14 +942,17 @@ public override unsafe void SetValue(object? component, object? value)
                 }
             }
 
-            if (hr == HRESULT.E_ABORT || hr == HRESULT.OLE_E_PROMPTSAVECANCELLED)
+            if (hr == HRESULT.S_OK || hr == HRESULT.S_FALSE)
             {
+                OnValueChanged(component, EventArgs.Empty);
+                _lastValue = value;
                 return;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2PropertyDescriptor.cs:949]: Does this need to be removed? There was no `return` for this case before.

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/IPictureDisp.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                dispatch->GetProperty(PInvoke.DISPID_PICT_HPAL, &variant).ThrowOnFailure();
+                paletteHandle = (OLE_HANDLE)variant.data.uintVal;
+            }
+
+            dispatch->GetProperty(PInvoke.DISPID_PICT_HANDLE, &variant).ThrowOnFailure();
+            OLE_HANDLE handle = (OLE_HANDLE)variant.data.uintVal;
+
+            dispatch->GetProperty(PInvoke.DISPID_PICT_WIDTH, &variant).ThrowOnFailure();
+            int width = variant.data.intVal;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/IPictureDisp.cs:48]: nit: `width `and `height` can be inline.

File: src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Ole32/IPictureTests.cs [hunk 4]
```diff
@@ -10,58 +10,58 @@
...
-            IPicture.Interface picture = MockAxHost.GetIPictureFromCursor(arrow.Handle);
-            Assert.NotNull(picture);
-            short type = picture.Type;
+            using var picture = IPicture.CreateFromIcon(Icon.FromHandle(arrow.Handle), copy: true);
+            Assert.False(picture.IsNull);
+            short type = picture.Value->Type;
             Assert.Equal((short)PICTYPE.PICTYPE_ICON, type);
 
-            int height = picture.Height;
+            int height = picture.Value->Height;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Ole32/IPictureTests.cs:25]: nit:  both `height `and `width `can be inline here too.

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/IPictureDisp.cs (hunk 3 above)
Comment (dreddy-work) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/IPictureDisp.cs:48]: or `GetProperty  ` expected to update `variant`?

Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/IPictureDisp.cs:48]: Yes, it is updating the VARIANT.

File: src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Ole32/IPictureTests.cs (hunk 4 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Ole32/IPictureTests.cs:25]: That's a judgement call that I don't think has much value here.

**Pull request merged**